plugins/rabbitmq/src/main/java/com/navercorp/pinpoint/plugin/rabbitmq/client/interceptor/ChannelBasicPublishInterceptor.java

```diff
@@ -162,16 +162,16 @@ private boolean validate(Object target, Object[] args) {
         if (args == null || args.length < 6) {
             return false;
         }
-        if (!(args[0] instanceof String)) {
+        if (args[0] !=null && !(args[0] instanceof String)) {
             return false;
         }
-        if (!(args[1] instanceof String)) {
+        if (args[1] !=null && !(args[1] instanceof String)) {
             return false;
         }
-        if (!(args[4] instanceof AMQP.BasicProperties)) {
+        if (args[4] !=null && !(args[4] instanceof AMQP.BasicProperties)) {
             return false;
         }
-        if (!(args[5] instanceof byte[])) {
+        if (args[5] !=null && !(args[5] instanceof byte[])) {
```

Comment:
I think this whole check for `args[5]` can be removed since it's not used at all in the interceptor.
What do you think?

Reply:
@Xylus In my test only args[4] instanceof AMQP.BasicProperties will be null. Since I did not know whether other args maybe null then I add check for all args. I know your concerns, for any args maybe null, the validation failure is easy to be ignored by user. If in some special case args[5] or [1] is null, exception will be printed into log which may cause we found the bugger more easier and quicker. So it depends what you prefer.
At least, arg[4] should be check for null.

Reply:
@jiaqifeng I definitely agree with you that *args[4]* should be allowed to be null.

But regarding *args[1]* and *args[5]*, I think it's important to remember that validation check in interceptor is done to see if the parameters won't throw exceptions in the interceptor code **only**, and not the original method.
For example, if *args[1]* (routing key) is null, **AMQImpl$Basic$Publish** constructor will throw an **IllegalStateException** if it is null, but I don't think it's the interceptor's responsibility to tell the user that *args[1]* should not be null. (This is more to do with the comment above regarding *args[1]*)
Similarly, I don't think we need to do any checks for *args[5]* at all in the interceptor as we never access it to do anything in the interceptor code, but do let me know if you think the check is still useful.

Reply:
@Xylus I agree with you. For I am not working on rabbitmq plugin for several months and forgot most details of rabbitmq client pacage's code.

Reply:
@jiaqifeng I think this part of your commit is just fine and can be merged as is :+1: 
Could you take a look at the other [comment](https://github.com/naver/pinpoint/pull/3996#discussion_r179973520)? That one does require a change I think.

Reply:
@Xylus 
As other comment, args[0] and args[1] are exchange and routingKey which will be check not to be null by ampq-client 3.5.5, then we do not need to check them against null. For args[4], which will breake the tx chain if null, so it shoudl be check against null. For args[5] as body which is alowed to be null in amqp-client 3.5.5 code, and it is not used in pinpoint code so I suggest to ignore it. Do you agree this solution? If yes, I will update the branch.

Reply:
@Xylus 
I decided to remove check for args[5] and updated the branch already.